itests/steps/autoscaler.py

```diff
@@ -339,19 +372,25 @@ def autoscaler_runs(context):
         except AttributeError:
             pass
 
-    if not hasattr(context, 'exception'):
+    if not once_only and not hasattr(context, 'exception'):
         # run it a second time to make sure nothing's changed
         context.autoscaler.run()
 
 
+@behave.when('the cluster has recently lost capacity')
+def lost_capacity(context):
+    context.mock_cluster_connector.return_value.get_removed_nodes_before_last_reload.return_value = [
+        KubernetesNode()
+    ]
+
+
 @behave.then('the autoscaler should scale rg(?P<rg>[12]) to (?P<target>\d+) capacity')
 def rg_capacity_change(context, rg, target):
     groups = list(context.autoscaler.pool_manager.resource_groups.values())
     if int(target) != groups[int(rg) - 1].target_capacity:
         assert_that(
             groups[int(rg) - 1].modify_target_capacity.call_args_list,
-            contains(
-                mock.call(int(target), dry_run=False),
+            has_item(
```

Comment:
FYI: I had to change this because sometimes we now only have one call to `modify_target_capacity`.